packages/salesforcedx-utils-vscode/src/commands/commandletExecutors.ts

```diff
@@ -159,9 +168,16 @@ export abstract class LibraryCommandletExecutor<T>
       const success = await vscode.window.withProgress(
         {
           title: nls.localize('progress_notification_text', this.executionName),
-          location: vscode.ProgressLocation.Notification
+          location: vscode.ProgressLocation.Notification,
+          cancellable: this.cancellable
         },
-        () => this.run(response)
+        (progress, token) => {
+          token.onCancellationRequested(() => {
+            this.cancelled = true;
+            notificationService.showCanceledExecution(this.executionName);
+          });
+          return this.run(response, progress, token);
```

Comment:
Bringing up the conversation from before:

Correct me if I'm wrong, but this is still dependent on how `run` is implemented and it always will. There is no way to stop this promise unless the concrete class resolves run appropriately when cancellation is requested. If i don't resolve run after cancellation is requested in my concrete class, the progress notification will keep going and we'll also show the cancellation message, even though we'll still be waiting for `withProgress` to finish.

Nothing is wrong with the code here, just setting the expectation of this behavior.

Reply:
Yeah, that's basically it! This logic will only handle displaying the appropriate notification to the user, it does not affect the actual execution of the command in `run`. The only difference is that the progress notification actually does not keep going after the user hits cancel, it disappears and the `Cancel` notification is displayed.

Reply:
Gotcha. Sounds like there's some internal behavior in withProgress that handles stopping the notification if the token is cancelled.